src/main/java/net/minestom/server/data/DataImpl.java

```diff
@@ -61,7 +61,13 @@ public boolean isEmpty() {
     @NotNull
     @Override
     public DataImpl clone() {
-        return (DataImpl) super.clone();
+        DataImpl newData = new DataImpl();
```

Comment:
Should still use `super.clone()` for polymorphism

Reply:
Well that removes the actual cloning as the super implementation just returns the same value.

Reply:
The `super.clone()` just returns a reference to the same object, so modifying the returned DataImpl causes updates to the object that was cloned which defeats the purpose of a clone method.

Reply:
After doing a bit more digging, I think this is due to the default `clone()` for HashMaps being a shallow copy, causing the DataImpl to get identical references rather than a deep clone of the ConcurrentHashMaps.